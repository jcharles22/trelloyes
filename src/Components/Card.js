import React from 'react';
import './Card.css'

function List(prop) {
    return (
        
        <div className="Card">
            <button>delete</button>
            <h3>{prop.title}</h3>
            <p>{prop.content}</p>
        </div>
        
    )
}

export default List;