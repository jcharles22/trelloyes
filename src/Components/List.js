import React from 'react';
import Card from './Card';
import './List.css';

function List(prop) {
    return (
        <section className="List">
            <header class="List-header">
                <h2>{prop.header}</h2>
            </header>
            <div className="List-cards"> 
                {prop.cards.map((card) => (
                    <Card
                    key={card.id}
                    title={card.title}
                    content={card.content}
                    />
                )
            )}
            <button>
                + Add Random Card
            </button>
            </div>
        </section>
    )
}

export default List;