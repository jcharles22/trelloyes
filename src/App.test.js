import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Components/Card';
import List from './Components/List';

describe('Card component', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render( <Card/>, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Card
        key={2}
        title={'first'}
        content={'lorem ipsum'}
        />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});

describe('List component', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render( <List/>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        
        const tree = renderer
          .create(<List
            key={2}
            header={"first"}
            cards={['a']} 
            />
           )
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
    });
