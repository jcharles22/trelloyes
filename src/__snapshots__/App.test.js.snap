// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button>
    delete
  </button>
  <h3>
    first
  </h3>
  <p>
    lorem ipsum
  </p>
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    class="List-header"
  >
    <h2>
      first
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button>
        delete
      </button>
      <h3 />
      <p />
    </div>
    <button>
      + Add Random Card
    </button>
  </div>
</section>
`;
